x/config/chooser_test.go

```diff
@@ -0,0 +1,389 @@
+				require.True(t, ok, "config has outbound")
+
+				require.NotNil(t, outbound.Unary, "must have unary outbound")
+				unary, ok := outbound.Unary.(*yarpctest.FakeOutbound)
+				require.True(t, ok, "unary outbound must be fake outbound")
+
+				transports := unary.Transports()
+				require.Equal(t, len(transports), 1, "must have one transport")
+
+				transport, ok := transports[0].(*yarpctest.FakeTransport)
+				require.True(t, ok, "must be a fake transport")
+				assert.Equal(t, transport.NopOption(), ":1234", "transport configured")
+
+				require.NotNil(t, unary.Chooser(), "must have chooser")
+				chooser, ok := unary.Chooser().(*peer.BoundChooser)
+				require.True(t, ok, "unary chooser must be a bound chooser")
+
+				fmt.Printf("U: %+v\n", chooser.Updater())
+				updater, ok := chooser.Updater().(*yarpctest.FakePeerListUpdater)
+				require.True(t, ok, "updater is a peer list updater")
+				assert.True(t, updater.Watch(), "peer list updater configured to watch")
+
+				list, ok := chooser.ChooserList().(*yarpctest.FakePeerList)
+				require.True(t, ok, "list is a fake peer list")
+				_ = list
+
+				dispatcher := yarpc.NewDispatcher(c)
+				assert.NoError(t, dispatcher.Start(), "error starting")
+				assert.NoError(t, dispatcher.Stop(), "error stopping")
+			},
+		},
+		{
+			desc: "use static peers with round robin and exercise choose",
+			given: whitespace.Expand(`
+				outbounds:
+					their-service:
+						unary:
+							fake-transport:
+								round-robin:
+									peers:
+									- 127.0.0.1:8080
+									- 127.0.0.1:8081
+			`),
+			test: func(t *testing.T, c yarpc.Config) {
+				outbound := c.Outbounds["their-service"]
+				unary := outbound.Unary.(*yarpctest.FakeOutbound)
+				transport := unary.Transports()[0].(*yarpctest.FakeTransport)
+				chooser := unary.Chooser().(*peer.BoundChooser)
+				binder := chooser.Updater()
+				list, ok := chooser.ChooserList().(*roundrobin.List)
+				require.True(t, ok, "chooser least pending")
+				_ = list
+
+				// Attempt to choose a peer
+				dispatcher := yarpc.NewDispatcher(c)
+				require.NoError(t, dispatcher.Start(), "error starting dispatcher")
+				defer func() {
+					require.NoError(t, dispatcher.Stop(), "error stopping dispatcher")
+				}()
+
+				// TODO https://github.com/yarpc/yarpc-go/issues/968
+				//require.True(t, dispatcher.IsRunning(), "dispatcher is running")
+				require.True(t, transport.IsRunning(), "transport is running")
+				require.True(t, unary.IsRunning(), "outbound is running")
+				require.True(t, list.IsRunning(), "chooser is running")
+				require.True(t, binder.IsRunning(), "binder is running")
+
+				ctx := context.Background()
+				ctx, cancel := context.WithTimeout(ctx, time.Second)
+				defer cancel()
+				peer, onFinish, err := chooser.Choose(ctx, nil)
+				require.NoError(t, err, "error choosing peer")
+				defer onFinish(nil)
+
+				assert.Equal(t, peer.Identifier(), "127.0.0.1:8080", "chooses first peer")
+			},
```

Comment:
these tests are a bit verbose to read, would be cool if they could be broken down into assertion components.  Not sure if possible, I'll leave it up to you.